core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -71,6 +72,7 @@ public String partition() {
   private final PartitionSet deleteFilePartitions;
   private final PartitionSet dropPartitions;
   private final CharSequenceSet deletePaths = CharSequenceSet.empty();
+  private final FilesToDeleteHolder filesToDelete = new FilesToDeleteHolder();
   private Expression deleteExpression = Expressions.alwaysFalse();
   private long minSequenceNumber = 0;
   private boolean hasPathOnlyDeletes = false;
```

Comment:
Can we get rid of `hasPathOnlyDeletes` now? We only use `deletePath` for path-only deletes after this PR. We can simply check if that set is empty.

Reply:
yep we can surely remove `hasPathOnlyDeletes`. I was trying to keep changes to an absolute minimum in this class and leave such non-essential refactorings out of the diff but I'm fine doing it here